ush/python/pygfs/jedi/jedi.py

```diff
@@ -0,0 +1,275 @@
+
+        Parameters
+        ----------
+        task_config : AttrDict
+            Dictionary of all configuration variables associated with a GDAS task.
+        algorithm (optional) : str
+            Name of the algorithm used to generate the JEDI configuration dictionary.
+            It will override the algorithm set in the task_config.JCB_<>_YAML file.
+
+        Returns
+        ----------
+        None
+        """
+
+        if 'JCB_BASE_YAML' in task_config.keys():
+            # Step 1: Fill templates of the JCB base YAML file
+            jcb_config = parse_j2yaml(task_config.JCB_BASE_YAML, task_config)
+
+            # Step 2: If algorithm is present then override the algorithm in the JEDI
+            #         config. Otherwise, if the algorithm J2-YAML is present, fill
+            #         its templates and merge.
+            if algorithm:
+                jcb_config['algorithm'] = algorithm
+            elif 'JCB_ALGO' in task_config.keys():
+                jcb_config['algorithm'] = task_config.JCB_ALGO
+            elif 'JCB_ALGO_YAML' in task_config.keys():
+                jcb_algo_config = parse_j2yaml(task_config.JCB_ALGO_YAML, task_config)
+                jcb_config.update(jcb_algo_config)
+
+            # Step 3: Generate the JEDI YAML using JCB
+            self.config = render(jcb_config)
+        elif 'JEDIYAML' in task_config.keys():
+            # Generate JEDI YAML without using JCB
+            self.config = parse_j2yaml(task_config.JEDIYAML, task_config,
+                                       searchpath=self.j2tmpl_dir)
+        else:
+            raise KeyError(f"Task config must contain JCB_BASE_YAML or JEDIYAML")
+
+    @logit(logger)
+    def execute(self, task_config: AttrDict, aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
+        """Execute JEDI application
+
+        Parameters
+        ----------
+        task_config: AttrDict
+            Attribute-dictionary of all configuration variables associated with a GDAS task.
+        aprun_cmd: str
+            String comprising the run command for the JEDI executable.
+        jedi_args (optional): List
+            List of strings comprising optional input arguments for the JEDI executable.
+
+        Returns
+        ----------
+        jedi_config: AttrDict
+            Attribute-dictionary of JEDI configuration rendered from a template.
+        """
+
+        chdir(task_config.DATA)
+
+        exec_cmd = Executable(aprun_cmd)
+        exec_cmd.add_default_arg(self.exe)
+        if jedi_args:
+            for arg in jedi_args:
+                exec_cmd.add_default_arg(arg)
+        exec_cmd.add_default_arg(self.yaml)
+
+        try:
+            exec_cmd()
+        except OSError:
+            raise OSError(f"Failed to execute {exec_cmd}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
```

Comment:
```suggestion
            raise OSError(f"FATAL ERROR: Failed to execute {exec_cmd}")
        except Exception:
            raise WorkflowException(f"FATAL ERROR: An error occurred during execution of {exec_cmd}")
```
Errors should be raised as `FATAL ERROR`.